### PR TITLE
feat: display users and devices E2E identity verification badges

### PR DESCRIPTION
**Pull request by przemvs**
## Description
Epic PR for UI for E2EI.

Displaying verification shields.

Possible to display:
- Proteus
- Valid MLS
- Expire soon MLS certificate
- Expired MLS certificate
- Revoked MLS certificate

Now we have only prepared functionality to use new verification shields. Proper displaying will be prepared in this epic: https://wearezeta.atlassian.net/browse/WPB-2253.

## Checklist

- [x] PR has been self reviewed by the author;
- [x] Hard-to-understand areas of the code have been commented;
- [x] If it is a core feature, unit tests have been added;

### CODE REVIEW

File: src/script/components/Badges/Badges.tsx [hunk 1]
```diff
@@ -0,0 +1,162 @@
...
+};
+
+const title = (isMLSConversation = false): CSSProperties => ({
+  color: isMLSConversation ? 'var(--green-500)' : 'var(--blue-500)',
+  fontSize: '12px',
+  lineHeight: '14px',
+  marginRight: '4px',
+});
+
+export const Badges: React.FC<BadgesProps> = ({
```
Comment (atomrc) [inline src/script/components/Badges/Badges.tsx]: suggestion: I don't think `Badges` properly conveys the functionality here. How about naming it `VerificationBadge`?

File: src/script/components/panel/UserDetails.tsx [hunk 2]
```diff
@@ -94,12 +96,12 @@ export const UserDetailsComponent: React.FC<UserDetailsProps> = ({
...
-            className="panel-participant__head__verified-icon"
-            data-uie-name="status-verified-participant"
-          />
-        )}
+        {/*{isSelfVerified && user.is_verified && (*/}
+        {/*  <Icon.Verified*/}
+        {/*    className="panel-participant__head__verified-icon"*/}
+        {/*    data-uie-name="status-verified-participant"*/}
+        {/*  />*/}
+        {/*)}*/}
```
Comment (atomrc) [inline src/script/components/panel/UserDetails.tsx]: how about this?

**Review by atomrc — APPROVED** [review]
LGTM

File: src/script/components/panel/UserDetails.tsx (hunk 2 above)
Comment (przemvs) [reply src/script/components/panel/UserDetails.tsx]: I'll keep it to know where should be this icon and forgot to remove.

File: src/script/page/MainContent/panels/preferences/devices/components/DetailedDevice.tsx [hunk 3]
```diff
@@ -19,52 +19,40 @@
...
-  return (
-    <>
-      <h3 className="preferences-devices-model" data-uie-name="device-model">
-        {device.model}
-      </h3>
-
-      <p className="preferences-devices-id">
-        <strong>{t('preferencesDevicesId')}</strong>
+export const DetailedDevice: React.FC<DeviceProps> = ({device, fingerprint, showVerificationStatus = true}) => {
+  const isProteusVerified = true;
```
Comment (PatrykBuniX) [inline src/script/page/MainContent/panels/preferences/devices/components/DetailedDevice.tsx]: **question**: This one will be handled is some following PR i assume?

Comment (przemvs) [reply src/script/page/MainContent/panels/preferences/devices/components/DetailedDevice.tsx]: Yup, we have epic task for shields :)

**Review by PatrykBuniX — APPROVED** [review]

File: src/script/components/VerificationBadges/VerificationBadges.tsx [hunk 4]
```diff
@@ -0,0 +1,161 @@
...
+
+  const showProteusBadge = conversationHasProtocol
+    ? conversationProtocol === ConversationProtocol.PROTEUS && isProteusVerified
+    : isProteusVerified;
+
+  return (
+    <div className="conversation-badges" style={{display: 'flex', alignItems: 'center', gap: '6px'}}>
+      {showMLSBadge && (
+        <div style={badgeWrapper}>
+          {displayTitle && <span style={title(true)}>Verified (End-to-end Identity)</span>}
```
Comment (PatrykBuniX) [inline src/script/components/VerificationBadges/VerificationBadges.tsx]: Hardcoded string spotted 👀, it should be added to translations file i believe, same thing for https://github.com/wireapp/wire-webapp/pull/15575/files#diff-48eda40ee53d9abb4cc638196bd62cac3b88ec9fbc209e69f359c892a7692469R147

File: src/script/page/MainContent/panels/preferences/DevicesPreferences/components/DeviceDetailsPreferences/DeviceDetailsPreferences.tsx [hunk 5]
```diff
@@ -139,9 +150,16 @@ const DeviceDetailsPreferences: React.FC<DevicesPreferencesProps> = ({
           </div>
         </section>
 
+        <section className="preferences-section">
+          <header className="preferences-header">
+            <hr className="preferences-separator" />
+          </header>
+        </section>
```
Comment (PatrykBuniX) [inline src/script/page/MainContent/panels/preferences/DevicesPreferences/components/DeviceDetailsPreferences/DeviceDetailsPreferences.tsx]: What's this one? New section for just a separator? 🤔

File: src/script/util/TypePredicateUtil.ts [hunk 6]
```diff
@@ -43,3 +44,8 @@ export function isConversationEntity(conversation: any): conversation is Convers
 export function isClientRecord(record: any): record is ClientRecord {
   return !!record.meta;
 }
+
+// TODO: Temporary checking for mls_public_keys
+export function hasMlsPublicKeys(client: any): client is RegisteredClient {
+  return !!client && 'mls_public_keys' in client;
+}
```
Comment (PatrykBuniX) [inline src/script/util/TypePredicateUtil.ts]: We should improve this guard.

File: src/script/components/VerificationBadges/VerificationBadges.tsx (hunk 4 above)
Comment (przemvs) [reply src/script/components/VerificationBadges/VerificationBadges.tsx]: Fixed ;)

File: src/script/page/MainContent/panels/preferences/DevicesPreferences/components/DeviceDetailsPreferences/DeviceDetailsPreferences.tsx (hunk 5 above)
Comment (przemvs) [reply src/script/page/MainContent/panels/preferences/DevicesPreferences/components/DeviceDetailsPreferences/DeviceDetailsPreferences.tsx]: This now is changed, now we have only separator, don't have whole section :)

File: src/script/components/panel/UserDetails.test.tsx [hunk 7]
```diff
@@ -54,20 +53,6 @@ describe('UserDetails', () => {
...
-    otherParticipant.devices.push(verifiedClient);
-
-    const props = {isGroupAdmin: true, isSelfVerified: true, participant: otherParticipant};
-
-    const {queryByTestId} = render(<UserDetails {...props} />);
-
-    expect(queryByTestId('status-verified-participant')).not.toBeNull();
-    expect(queryByTestId('status-admin')).not.toBeNull();
-  });
-
```
Comment (przemvs) [inline src/script/components/panel/UserDetails.test.tsx:70]: This part is unnecessary, we have now VerificationBadges component for this.

File: src/script/components/userDevices/DeviceCard.test.tsx [hunk 8]
```diff
@@ -37,36 +37,6 @@ function createClientEntity(clientEntity: Partial<ClientEntity>): ClientEntity {
...
-        meta: {
-          isVerified: ko.observable<boolean>(false),
-        },
-      }),
-      showIcon: true,
-    };
-
-    const {getByTestId} = render(<DeviceCard {...props} />);
-    expect(getByTestId('status-desktop-device')).not.toBeNull();
-  });
```
Comment (przemvs) [inline src/script/components/userDevices/DeviceCard.test.tsx:53]: We don't display desktop icon now.

image

File: src/script/components/userDevices/DeviceCard.test.tsx [hunk 9]
```diff
@@ -37,36 +37,6 @@ function createClientEntity(clientEntity: Partial<ClientEntity>): ClientEntity {
...
-        meta: {
-          isVerified: ko.observable<boolean>(false),
-        },
-      }),
-      showIcon: true,
-    };
-
-    const {getByTestId} = render(<DeviceCard {...props} />);
-    expect(getByTestId('status-mobile-device')).not.toBeNull();
-  });
```
Comment (przemvs) [inline src/script/components/userDevices/DeviceCard.test.tsx:68]: We don't display mobile icon now.

image

File: src/script/components/userDevices/DeviceCard.test.tsx [hunk 10]
```diff
@@ -82,36 +52,4 @@ describe('DeviceCard', () => {
...
-          isVerified: ko.observable<boolean>(false),
-        },
-      }),
-      showIcon: true,
-      showVerified: true,
-    };
-
-    const {getByTestId} = render(<DeviceCard {...props} />);
-    expect(getByTestId('user-device-not-verified')).not.toBeNull();
-  });
```
Comment (przemvs) [inline src/script/components/userDevices/DeviceCard.test.tsx:116]: Verified Icon displaying now in VerificationBadges.

File: src/script/util/certificateDetails.ts [hunk 11]
```diff
@@ -0,0 +1,74 @@
...
+ * along with this program. If not, see http://www.gnu.org/licenses/.
+ *
+ */
+
+import * as x509 from '@peculiar/x509';
+
+import {MLSStatuses} from 'Components/VerificationBadges';
+
+// TODO: Will be changed when I get information how much hours before we need to display this status.
+const EXPIRATION_HOURS = 4;
```
Comment (przemvs) [inline src/script/util/certificateDetails.ts]: We still waiting for information.

**Review by PatrykBuniX — APPROVED** [review]

**Pull request merged**